pupil_src/shared_modules/video_capture/ndsi_backend.py

```diff
@@ -408,76 +433,83 @@ def re_build_ndsi_menu(self):
             ui.Info_Text("Pupil Mobile Commspec v{}".format(__protocol_version__))
         )
 
-        def host_selection_list():
-            devices = {
-                s["host_uuid"]: s["host_name"]  # removes duplicates
-                for s in self.network.sensors.values()
-            }
-            devices = [pair for pair in devices.items()]  # create tuples
-            # split tuples into 2 lists
-            return zip(*(devices or [(None, "No hosts found")]))
-
-        def view_host(host_uuid):
-            if self.selected_host != host_uuid:
-                self.selected_host = host_uuid
-                self.re_build_ndsi_menu()
-
-        host_sel, host_sel_labels = host_selection_list()
+        host_sel, host_sel_labels = self.host_selection_list()
         ui_elements.append(
             ui.Selector(
                 "selected_host",
                 self,
                 selection=host_sel,
                 labels=host_sel_labels,
-                setter=view_host,
+                setter=self.view_host,
                 label="Remote host",
             )
         )
 
         self.menu.extend(ui_elements)
+
+        self.menu.append(
+            ui.Button("Auto select in all processes", self.auto_select_manager)
+        )
+
```

Comment:
Can this be abstracted into a `base_backend` function? This way would not have to change the button text n multiple locations if needed to.